app/controllers/cloud_subnet_controller.rb

```diff
@@ -12,4 +12,253 @@ class CloudSubnetController < ApplicationController
+    # Todo(gildub): Replace with angular lookup to narrow choice dynamically
+    @network_choices = {}
+    CloudNetwork.all.each { |network| @network_choices[network.name] = network.ems_ref }
+    @cloud_tenant_choices = {}
+    CloudTenant.all.each { |tenant| @cloud_tenant_choices[tenant.name] = tenant.id }
+    drop_breadcrumb(
+      :name => _("Add New %{model}") % {:model => ui_lookup(:table => 'cloud_subnet')},
+      :url  => "/cloud_subnet/new"
+    )
+  end
+
+  def create
+    assert_privileges("cloud_subnet_new")
+    case params[:button]
+    when "cancel"
+      javascript_redirect :action => 'show_list',
+                          :flash_msg => _("Add of new %{model} was cancelled by the user") % {:model => ui_lookup(:table => 'cloud_subnet')}
+
+    when "add"
+      @subnet = CloudSubnet.new
+      options = form_params
+      cloud_tenant = find_by_id_filtered(CloudTenant, options[:cloud_tenant_id])
+      ems = ExtManagementSystem.find(options[:ems_id])
+      valid_action, action_details = CloudSubnet.validate_create_subnet(ems)
+      if valid_action
+        begin
+          CloudSubnet.create_subnet(ems, options)
+          # Todo: To replace with targeted refresh when avail. or either use tasks
+          EmsRefresh.queue_refresh(ManageIQ::Providers::NetworkManager)
+          add_flash(_("Creating %{subnet} \"%{subnet_name}\"") % {
+            :subnet      => ui_lookup(:table => 'cloud_subnet'),
+            :subnet_name => options[:name]})
+        rescue => ex
+          add_flash(_("Unable to create %{subnet} \"%{subnet_name}\": %{details}") % {
+            :subnet      => ui_lookup(:table => 'cloud_subnet'),
+            :subnet_name => options[:name],
+            :details     => ex}, :error)
+        end
+        @breadcrumbs.pop if @breadcrumbs
+        session[:flash_msgs] = @flash_array.dup if @flash_array
+        javascript_redirect :action => "show_list"
+      else
+        @in_a_form = true
+        add_flash(_(action_details), :error) unless action_details.nil?
+        drop_breadcrumb(
+          :name => _("Add New %{model}") % {:model => ui_lookup(:table => 'cloud_subnet')},
+          :url  => "/cloud_subnet/new"
+        )
+        javascript_flash
+      end
+    end
+  end
+
+  def delete_subnets
+    assert_privileges("cloud_subnet_delete")
+
+    subnets = if @lastaction == "show_list" || (@lastaction == "show" && @layout != "cloud_subnet")
+                find_checked_items
+              else
+                [params[:id]]
+              end
+
+    if subnets.empty?
+      add_flash(_("No %{models} were selected for deletion.") % {
+        :models => ui_lookup(:tables => "cloud_subnet")
+      }, :error)
+    end
+
+    subnets_to_delete = []
+    subnets.each do |s|
+      subnet = CloudSubnet.find_by_id(s)
+      if subnet.nil?
+        add_flash(_("%{model} no longer exists.") % {:model => ui_lookup(:table => "cloud_subnet")}, :error)
```

Comment:
I'd just put _('Cloud subnet no longer exists.') here. No benefit from using `ui_lookup` here and we want to get rid or that.

The same applies above.